Sources/Paywalls/Components/Transitions/PaywallAnimation.swift

```diff
@@ -0,0 +1,94 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PaywallAnimation.swift
+//
+//  Created by Jacob Zivan Rakidzich on 8/20/25.
+// swiftlint:disable missing_docs
+
+import Foundation
+
+public extension PaywallComponent {
+    struct Animation: PaywallComponentBase {
+        public let type: AnimationType
+        public let msDelay: Int?
+        public let msDuration: Int?
+
+        init(type: AnimationType, msDelay: Int?, msDuration: Int?) {
+            self.type = type
+            self.msDelay = msDelay
+            self.msDuration = msDuration
+        }
+
+        public init(from decoder: any Decoder) throws {
+            let passthrough = try AnimationCodingContainer(from: decoder)
+
+            self.type = AnimationType.from(passthrough.type) ?? .easeInOut
+            self.msDelay = passthrough.msDelay
+            self.msDuration = passthrough.msDuration
+        }
+
+        public func encode(to encoder: any Encoder) throws {
+            try AnimationCodingContainer(type: type.codingContainer, msDelay: msDelay, msDuration: msDuration)
+                .encode(to: encoder)
+        }
+    }
+
+    internal struct AnimationCodingContainer: Codable {
```

Comment:
Same question about `CodingContainer` vs using `CodingKeys` like in the other components